src/xrGame/PhraseDialog.cpp

```diff
@@ -254,6 +254,9 @@ CPhrase* CPhraseDialog::AddPhrase(
 {
     CPhrase* phrase = NULL;
     CPhraseGraph::CVertex* _vertex = data()->m_PhraseGraph.vertex(phrase_id);
+
+    VERIFY2(!_vertex, make_string("Dublicate phrase ID: [%s] for phrase: [%s]. Existed phrase by this ID: [%s]", phrase_id.c_str(), text, _vertex->data()->GetText()));
+
     if (!_vertex)
```

Comment:
Should this check then also be removed since `_vertex` should essentially always be null?

Reply:
This mistake can easily be made when a mod developer creates a complex dialect from scripts.

Reply:
On [256](https://github.com/OpenXRay/xray-16/pull/1555/files#diff-5a47ae2a8b262938a35d803b14650aa1f50a585a87f2917c138e4c92eae43333R256) line `_vertex` will be null when adding a new phrase by ID. But if a phrase has already been added for the same ID, then the `_vertex` will not be null. This means that a phrase has already been added for this ID. And this behavior is a bug. And this `VERIFY` will appear only in the debug build.
Or are you talking about replacing the `VERIFY` on `R_ASSERT`?

Reply:
Yes I do understand how `_vertex` _can_ be null.

I was talking about the if statement right under that for which I recommended removing it since AFTER the `VERIFY` you added, `_vertex` should never be null and in release mode the compiler will actually assume that and likely remove the check.

But since adding a duplicate id triggers an assertions later I think it's fine to leave it as `VERIFY` although using `R_ASSERT` should also be completely fine.

Reply:
@Xottab-DUTY what is R_ASSERT_CURE? I haven't seen such a macro